### PR TITLE
add authentication week project brief

### LINKED ISSUE #350: authentication week project
### Planning on gitter
image
image

### PR DESCRIPTION
**Pull request by jsms90**
Add task, requirements & possible stretch goals.

Still to come - examples to help students with ideas (requested in other weeks)

fixes #350

### CODE REVIEW

**Comment by ronanyeah** [comment]
I think that that is too specific. We just want them to use auth. You could use auth to make a game, a todo app, a [twitter clone](http://canaryapp.herokuapp.com/) :heart: or whatever. Do we really want 4 CMSs? As long as there are protected routes and secure cookies (and perhaps multiple access levels), they have successfully absorbed Week 7.

**Comment by sohilpandya** [comment]
Too specific. We should let there brain wander and see where it leads as long as needs some kind of authentication

**Comment by jsms90** [comment]
Ooo ok!! Yeah, they've basically done something like this a few times now anyway.

@ronanyeah @sohilpandya What exactly do you want me to do? Remove the task description altogether and leave the requirements? Change the requirements?

File: coursebook/week-7/project.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## Project
+### Task
+Create a CMS (Content Management System) - a website which allows you to manage your site's content without any technical training. But what does "managing" your content mean? Essentially, this boils down to performing CRUD operations (CREATE, READ, UPDATE & DELETE) on your site's content.
```
Comment (ronanyeah) [inline coursebook/week-7/project.md]: I think we can lose this.

File: coursebook/week-7/project.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+You are free to create any kind of site that you like, so get creative! But if your team is really stuck, there are [some examples below](#examples).
```
Comment (ronanyeah) [inline coursebook/week-7/project.md]: This is good, maybe just add in 'as long as it fits the requirements'.

Comment (jsms90) [inline coursebook/week-7/project.md]: done :wink:

File: coursebook/week-7/project.md (hunk 1 above)
Comment (jsms90) [inline coursebook/week-7/project.md]: :+1:

File: coursebook/week-7/project.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+You should create your site from a user-centred design perspective. So when your team decides what you want to build, make sure you start your planning phase with user stories. You want to make sure that you are building the most important features first :tad
+
+### Requirements
++ [ ] Login form with 2 fields - username and password
++ [ ] Client-side _and_ server-side validation on login form, including error handling that provides feedback to users
++ [ ] Users only have to log in once (i.e. implement a cookie-based session on login)
++ [ ] Username is visible on each page of the site after logging in
++ [ ] Authenticated users should be able to add new content (CREATE) & edit the content that they have submitted (UPDATE).
++ [ ] An unauthenticated user should be able to view the content (i.e. READ)
++ [ ] Name of the author should be visible against each post
```
Comment (ronanyeah) [inline coursebook/week-7/project.md]: The above 3 aren't needed now. We could replace with:
- There should be protected routes and unprotected routes that depend on the user having a cookie or not (or what level of access they have).

**Comment by ronanyeah** [comment]
Examples are good, such as the legendary [Canary](http://canaryapp.herokuapp.com/).

**Comment by ronanyeah** [comment]
The requirements are good!

File: coursebook/week-7/project.md (hunk 3 above)
Comment (jsms90) [inline coursebook/week-7/project.md]: "Name of the author" is still good, no? You want each "tweet" in canary to tell you which person posted it, right?

Comment (ronanyeah) [inline coursebook/week-7/project.md]: It's only valid for apps that you can view other peoples content in.

File: coursebook/week-7/project.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Make sure you're creating your site from a user-centred design perspective, by starting your planning phase with user stories. You always want to be building the most important features first.
+
+The only limits are that your project must fulfill the requirements below.
+
+### Requirements
++ [ ] Login form with 2 fields - username and password
++ [ ] Client-side _and_ server-side validation on login form, including error handling that provides feedback to users
++ [ ] Users only have to log in once (i.e. implement a cookie-based session on login)
++ [ ] Username is visible on each page of the site after logging in
++ [ ] Name of the author should be visible against each post
```
Comment (eliascodes) [inline coursebook/week-7/project.md]: If you remove the CMS theme, I think you should also remove references to posts? Maybe something like:
> Any user-submitted content should be labelled with the authors username

File: coursebook/week-7/project.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+### Task
+
+<p style="text-align: center;">You are free to create any kind of site that you like, so get creative!</p>
+
+Make sure you're creating your site from a user-centred design perspective, by starting your planning phase with user stories. You always want to be building the most important features first.
+
+The only limits are that your project must fulfill the requirements below.
+
+### Requirements
++ [ ] Login form with 2 fields - username and password
```
Comment (eliascodes) [inline coursebook/week-7/project.md]: What about signup?

File: coursebook/week-7/project.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
++ [ ] There should be protected routes and unprotected routes that depend on the user having a cookie or not (or what level of access they have).
++ [ ] Website content should be stored in a database
+
+Note:
++ Client-side and server-side validation on content posting is optional
++ Whether you also allow users to delete the content that they have posted will depend on the project you decide to create.
+
+#### Possible stretch goals
+Roles and permissions:
++ [ ] Have an "admin" level user (role) who can edit and delete all posts :scream: (permissions)
```
Comment (eliascodes) [inline coursebook/week-7/project.md]: More references to "posts" here and below. Could just replace with "content"?

**Comment by eliascodes** [comment]
Also the `<p style="...">` tag is redundant, it gets stripped by the GFM parser.

File: coursebook/week-7/project.md (hunk 5 above)
Comment (jsms90) [inline coursebook/week-7/project.md]: :thinking: I dunno. Maybe. I did think about it, but it seems odd to enable people to register themselves with no verification that they are who they say they are.

If it's just about submitting content to their database, we've already got them posting new content for that.

Comment (eliascodes) [inline coursebook/week-7/project.md]: OK, I think that's fine.

**Pull request merged**